### PR TITLE
chore(libp2p): expose `json` feature from `request-response`

### PR DESCRIPTION
**Pull request by radiohertz**
## Description

`cbor` feature was exposed in `libp2p` but not `json`.

## Change checklist

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] A changelog entry has been made in the appropriate crates

### CODE REVIEW

**Review by mxinden — COMMENTED** [review]
Good catch. Mind adding a changelog entry in `libp2p/CHANGELOG.md`?

See https://github.com/libp2p/rust-libp2p/blob/master/docs/release.md#make-changelog-entries for details.

**Comment by radiohertz** [comment]
@mxinden done!

File: libp2p/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 0.52.2 - unreleased
+
+- Add `json` feature which exposes `request_response::json`
```
Comment (thomaseizinger) [inline libp2p/CHANGELOG.md]:
```suggestion
- Add `json` feature which exposes `request_response::json`.
```

**Review by thomaseizinger — APPROVED** [review]

File: libp2p/CHANGELOG.md (hunk 1 above)
Comment (thomaseizinger) [reply libp2p/CHANGELOG.md]: It seems I cannot commit to your PR.

File: libp2p/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
+## 0.52.2 - unreleased
+
+- Add `json` feature which exposes `request_response::json`.
```
Comment (thomaseizinger) [inline libp2p/CHANGELOG.md:6]: Sorry that I missed this earlier but this is missing a reference to the PR.

**Review by thomaseizinger — COMMENTED** [review]
One more thing (plus resolve merge conflicts please).

**Review by thomaseizinger — APPROVED** [review]
Thank you for your patience! :)

**Pull request merged**